tests/model/test_model.py

```diff
@@ -2079,20 +2098,73 @@ def test_toggle_stream_pinned_status(
                     "id": 0,
                 },
                 True,
+                False,
             ),
         ],
-        ids=["not_in_pm_narrow", "not_in_pm_narrow_with_sender", "start", "stop"],
+        ids=[
+            "not_in_pm_narrow",
+            "not_in_pm_narrow_with_sender",
+            "start",
+            "start when animating",
+            "stop",
+        ],
     )
-    def test__handle_typing_event(self, mocker, model, narrow, event, called):
+    def test__handle_typing_event(
+        self,
+        mocker,
+        model,
+        narrow,
+        event,
+        is_animation_in_progress,
+        is_recipient_actually_typing,
+    ):
         event["type"] = "typing"
 
-        mocker.patch("zulipterminal.ui.View.set_footer_text")
+        mocker.patch("zulipterminal.model.Model._show_typing_animation")
         model.narrow = narrow
         model.user_dict = {"hamlet@zulip.com": {"full_name": "hamlet"}}
+        model.is_recipient_typing = False
+        model.is_typing_animation_in_progress = is_animation_in_progress
 
         model._handle_typing_event(event)
 
-        assert model.controller.view.set_footer_text.called == called
+        assert model.is_recipient_typing == is_recipient_actually_typing
+        assert model._show_typing_animation.called == (
+            is_recipient_actually_typing and not is_animation_in_progress
+        )
+
+    @pytest.mark.parametrize(
+        "event, expected_indicator",
+        [
+            # In PM narrow with the sender, OP - 'start'
+            (
+                {"op": "start"},
+                True,
+            ),
+            # OP - 'stop'
+            (
+                {"op": "stop"},
+                False,
+            ),
+        ],
+        ids=["start", "stop"],
+    )
+    def test__show_typing_animation(self, mocker, model, event, expected_indicator):
+        mocker.patch("zulipterminal.ui.View.set_footer_text")
+        recipient = {"full_name": "hamlet"}
+        model.is_recipient_typing = event["op"] == "start"
+        model.last_update_time = datetime.now() - timedelta(seconds=14.5)
```

Comment:
Could we have a test for `last_updated_time` > 15?

Reply:
I think this is already tested as the async function breaks when last_update_time crosses 15

Reply:
And also when `is_recipient_typing` is False it never enters, so breaking the loop because of `is_recipient_typing` is also tested.

Reply:
Right, I think this is fine for now.